controllers/componentdetectionquery_controller.go

```diff
@@ -320,11 +324,22 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 
 				devfilesMapReturned, devfilesURLMapReturned, dockerfileContextMapReturned, componentPortsMapReturned, revision, err = cdqanalysis.CloneAndAnalyze(k8sInfoClient, req.Namespace, req.Name, context, cdqInfo)
 				if err != nil {
-					log.Error(err, fmt.Sprintf("Error running cdq analysis... %v", req.NamespacedName))
+					switch err.(type) {
+					case *cdqanalysis.NoDevfileFound:
+						log.Error(err, fmt.Sprintf("NoDevfileFound error running cdq analysis... %v", req.NamespacedName))
+					case *cdqanalysis.NoDockerfileFound:
+						log.Error(err, fmt.Sprintf("NoDockerfileFound error running cdq analysis... %v", req.NamespacedName))
+					default:
```

Comment:
Are there any other user errors we need to consider here? What if the repo contains a devfile, but it's invalid or unreadable?

Reply:
add more user cases

Reply:
Do we need to consider this scenario too? https://github.com/redhat-appstudio/application-service/blob/15f897cd79f7df90e5247c89d7f3438196a3f5c6/cdq-analysis/pkg/componentdetectionquery.go#L197-L202

I tried to walk through Alizer, and it seemed there were a couple cases where it would return errors that would count as user errors (e.g. https://github.com/devfile/alizer/blob/main/pkg/apis/recognizer/component_recognizer.go#L288)

Reply:
for Alizer, we are calling `DetectComponentsWithSettings` so which calls `getComponentsWithoutConfigFile`, which won't return error if no components being detected
https://github.com/devfile/alizer/blob/6b0b633dc3423413e8677b9ac7de0d778020c9f8/pkg/apis/recognizer/component_recognizer.go#L117C9-L117C9